Realm/ObjectServerTests/SwiftMongoClientTests.swift

```diff
@@ -0,0 +1,1222 @@
+        XCTAssertEqual(resultObjectIds2.count, 3)
+    }
+
+    func testMongoCollectionDeleteManyAsyncAwait() async throws {
+        let collection = try await setupMongoCollection()
+
+        let document: Document = ["name": "tomas", "breed": "jack rusell"]
+        let document1: Document = ["name": "fito", "breed": "goberian"]
+        let document2: Document = ["name": "fosca", "breed": "labradoodle"]
+        let document3: Document = ["name": "balo", "breed": "pug"]
+
+        let deletedDocumentCount = try await collection.deleteManyDocuments(filter: document)
+        XCTAssertEqual(deletedDocumentCount, 0)
+
+        let objectIds = try await collection.insertMany([document, document1, document2, document3])
+        XCTAssertNotNil(objectIds)
+        XCTAssertEqual(objectIds.count, 4)
+
+        let deletedDocumentCount2 = try await collection.deleteManyDocuments(filter: document)
+        XCTAssertEqual(deletedDocumentCount2, 1)
+
+        let resultObjectIds2 = try await collection.find(filter: [:])
+        XCTAssertEqual(resultObjectIds2.count, 3)
+    }
+
+    func testMongoCollectionAggregateAsyncAwait() async throws {
+        let collection = try await setupMongoCollection()
+
+        let document: Document = ["name": "tomas", "breed": "jack rusell"]
+        let document1: Document = ["name": "lucas", "breed": "german shepard"]
+        let document2: Document = ["name": "fito", "breed": "goberian"]
+        let document3: Document = ["name": "balo", "breed": "pug"]
+
+        let objectIds = try await collection.insertMany([document, document1, document2, document3])
+        XCTAssertNotNil(objectIds)
+        XCTAssertEqual(objectIds.count, 4)
+
+        let documents = try await collection.aggregate(pipeline: [["$match": ["name": "tomas"]], ["$group": ["_id": "$name"]]])
+        XCTAssertNotNil(documents)
+        XCTAssertEqual(documents.count, 1)
+    }
+
+    func testMongoCollectionCountAsyncAwait() async throws {
+        let collection = try await setupMongoCollection()
+
+        let document: Document = ["name": "tomas", "breed": "jack rusell"]
+        let document1: Document = ["name": "lucas", "breed": "german shepard"]
+        let document2: Document = ["name": "fito", "breed": "goberian"]
+        let document3: Document = ["name": "balo", "breed": "pug"]
+
+        let objectIds = try await collection.insertMany([document, document1, document2, document3])
+        XCTAssertNotNil(objectIds)
+        XCTAssertEqual(objectIds.count, 4)
+
+        let count = try await collection.count(filter: [:])
+        XCTAssertEqual(count, 4)
+
+        let count2 = try await collection.count(filter: document)
+        XCTAssertEqual(count2, 1)
+
+        let count3 = try await collection.count(filter: [:], limit: 2)
+        XCTAssertEqual(count3, 2)
+
+        let count4 = try await collection.count(filter: document3, limit: 1)
+        XCTAssertEqual(count4, 1)
```

Comment:
I'm not sure this adds anything to the test.

Reply:
changed this to test limit option on deleteManyDocuments function